python-sdk/pyproject.toml

```diff
@@ -59,7 +59,7 @@ google = [
 ]
 snowflake = [
     "apache-airflow-providers-snowflake",
-    "snowflake-sqlalchemy>=1.2.0,<=1.2.4",
+    "snowflake-sqlalchemy>=1.4.3,<1.5.0",
```

Comment:
does it cause any problems if we increase the version from 1.2.0 to 1.4.3 and a customer has some other things  which depend on 1.2.0 in the env?